### PR TITLE
Update/v23

### PR DESCRIPTION
**Pull request by magik6k**
## Why does this PR exist?

We want v23 Groth parameters!

## What's in this PR?

This PR updates go-sectorbuilder to use v23 Groth parameters and includes updates to conform to new specs-actors code and filecoin-ffi API.

### CODE REVIEW

File: interface.go [hunk 1]
```diff
@@ -20,12 +19,12 @@ type Interface interface {
 	AcquireSectorNumber() (abi.SectorNumber, error)
 	Scrub([]abi.SectorNumber) []*Fault
 
-	GenerateEPostCandidates(sectorInfo ffi.SortedPublicSectorInfo, challengeSeed abi.PoStRandomness, faults []abi.SectorNumber) ([]ffi.PoStCandidateWithTicket, error)
-	GenerateFallbackPoSt(sectorInfo ffi.SortedPublicSectorInfo, challengeSeed abi.PoStRandomness, faults []abi.SectorNumber) ([]ffi.PoStCandidateWithTicket, []byte, error)
-	ComputeElectionPoSt(sectorInfo ffi.SortedPublicSectorInfo, challengeSeed abi.PoStRandomness, winners []abi.PoStCandidate) ([]byte, error)
+	GenerateEPostCandidates(sectorInfo []abi.SectorInfo, challengeSeed abi.PoStRandomness, faults []abi.SectorNumber) ([]ffi.PoStCandidateWithTicket, error)
+	GenerateFallbackPoSt(sectorInfo []abi.SectorInfo, challengeSeed abi.PoStRandomness, faults []abi.SectorNumber) ([]ffi.PoStCandidateWithTicket, []abi.PoStProof, error)
```
Comment (laser) [inline interface.go:23]: Note that these `Generate*` methods can return more than one PoSt proof.

File: sectorbuilder.go [hunk 2]
```diff
@@ -445,30 +444,22 @@ func sizeFromConfig(cfg Config) (abi.SectorSize, error) {
 
 func sizeFromProofType(p abi.RegisteredProof) (abi.SectorSize, error) {
 	switch p {
-	case abi.RegisteredProof_WinStackedDRG32GiBSeal:
-		return 1 << 35, nil
-	case abi.RegisteredProof_WinStackedDRG32GiBPoSt:
-		return 1 << 35, nil
 	case abi.RegisteredProof_StackedDRG32GiBSeal:
-		return 1 << 35, nil
+		return 32 << 30, nil
```
Comment (laser) [inline sectorbuilder.go]: bug!

Comment (magik6k) [reply sectorbuilder.go]: No, it should be 30 (2^30 = 1GiB)

Comment (magik6k) [reply sectorbuilder.go]: (the post one is wrong)

File: sectorbuilder.go [hunk 3]
```diff
@@ -445,30 +444,22 @@ func sizeFromConfig(cfg Config) (abi.SectorSize, error) {
 
 func sizeFromProofType(p abi.RegisteredProof) (abi.SectorSize, error) {
 	switch p {
-	case abi.RegisteredProof_WinStackedDRG32GiBSeal:
-		return 1 << 35, nil
-	case abi.RegisteredProof_WinStackedDRG32GiBPoSt:
-		return 1 << 35, nil
 	case abi.RegisteredProof_StackedDRG32GiBSeal:
-		return 1 << 35, nil
+		return 32 << 20, nil
```
Comment (acruikshank) [inline sectorbuilder.go]: It takes a fair bit of scrutiny and understanding to understand while we're shifting a power of 2 by a multiple of 10. Defining constants `MB := 1<<20` and `KB := 1<<10` and then returning `32*MB`, `2*KB`, `8*MB` etc. would clarify considerably.

**Review by acruikshank — DISMISSED** [review]
One suggestion, but looks good overall.

File: sectorbuilder.go (hunk 2 above)
Comment (laser) [reply sectorbuilder.go]: @magik6k 

> No, it should be 30 (2^30 = 1GiB)

Ah, OK. My bad.

We might as well just use the same expressions that exist in the [source-of-truth](https://github.com/filecoin-project/rust-fil-proofs/blob/master/filecoin-proofs/src/constants.rs#L11). Thoughts, @magik6k ?

**Review by magik6k — COMMENTED** [review]
Later changes LGTM

File: sectorbuilder.go [hunk 4]
```diff
@@ -444,31 +448,22 @@ func sizeFromConfig(cfg Config) (abi.SectorSize, error) {
 }
 
 func sizeFromProofType(p abi.RegisteredProof) (abi.SectorSize, error) {
```
Comment (magik6k) [inline sectorbuilder.go:450]: likely want to just use https://github.com/filecoin-project/specs-actors/pull/212/files once merged

**Pull request merged**